frontend/apps/reader/modules/readeruserhyph.lua

```diff
@@ -0,0 +1,248 @@
+        end
+    else
+        line = nil -- EOF
+    end
+
+    -- write new entry
+    if hyphenation and hyphenation~="" then
+        new_dict:write(string.format("%s;%s\n", word, hyphenation))
+    end
+
+    -- write old entry if there was one
+    if line then
+        new_dict:write(line .. "\n")
+    end
+
+    -- copy rest of file
+    repeat
+        line = dict:read()
+        if line then
+            new_dict:write(line .. "\n")
+        end
+    until (not line)
+
+    dict:close()
+    new_dict:close()
+    os.remove(dict_file)
+    os.rename(new_dict_file, dict_file)
+
+    self:loadUserDictionary(true) -- dictionary has changed, force a reload here
+end
+
+function ReaderUserHyph:modifyUserEntry(word)
+    if word:find("[ ,;-%.]") then return end -- no button if more than one word
+
+    if not self.ui.document then return end
+
+    local suggested_hyphenation = self.ui.document:getHyphenationForWord(word)
+
+    local input_dialog
+    input_dialog = InputDialog:new{
+        title = T(_("Hyphenation entry for: \"%1\""), word),
+        description = _("Add hyphenation positions with hyphens ('-') or spaces (' ')."),
+        input = suggested_hyphenation,
+        old_hyph = suggested_hyphenation,
+        input_type = "string",
+        buttons = {
+            {
+                {
+                    text = _("Cancel"),
+                    callback = function()
+                        UIManager:close(input_dialog)
+                    end,
+                },
+                {
+                    text = _("Remove"),
+                    callback = function()
+                        UIManager:close(input_dialog)
+                        self:updateDictionary(word)
+                    end,
+                },
+                {
+                    text = _("Store"),
+                    is_enter_default = true,
+                    callback = function()
+                        local new_suggestion = input_dialog:getInputText()
+                        new_suggestion = new_suggestion:gsub(" ","-") -- replace spaces with hyphens
+                        new_suggestion = new_suggestion:gsub("^-","") -- remove leading hypenations
+                        new_suggestion = new_suggestion:gsub("-$","") -- remove trailing hypenations
+
+                        if self:checkHyphenation(new_suggestion, word) then
+                            -- don't save if no changes
+                            if self.ui.document:getLowercasedWord(new_suggestion)
+                                ~= self.ui.document:getLowercasedWord(input_dialog.old_hyph) then
+                                self:updateDictionary(word, new_suggestion)
+                            end
+                            UIManager:close(input_dialog)
+                        else
+                            UIManager:show(InfoMessage:new{
+                                text = T(_("Wrong hyphenation!\nPlease check!"), self.dict_file),
```

Comment:
`Please check!` Are we that enthusiastic in other InfoMessage ? :)
Just "Invalid hyphenation." and letting the user on this dialog is enough for him to realize he has to check :)